src/figuresClasses.ts

```diff
@@ -1,11 +1,103 @@
-export interface Figure {}
+export interface Figure {
+  color: 'red' | 'green' | 'blue';
+  shape: 'triangle' | 'circle' | 'rectangle';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  color: 'red' | 'green' | 'blue';
+
+  shape: 'triangle';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All side lengths must be greater than 0.');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        `Sides ${a}, ${b}, and ${c} can't form a valid triangle.`,
+      );
+    }
+    this.color = color;
+    this.shape = 'triangle';
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  color: 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+  shape: 'circle';
```

Comment:
The `shape` property should be declared as `readonly` to match the `Figure` interface.